engine/execution/messages.go

```diff
@@ -16,11 +18,57 @@ type ComputationOrder struct {
 }
 
 type ComputationResult struct {
-	ExecutableBlock   *entity.ExecutableBlock
-	StateSnapshots    []*delta.SpockSnapshot
-	Events            []flow.Event
-	ServiceEvents     []flow.Event
-	TransactionResult []flow.TransactionResult
-	GasUsed           uint64
-	StateReads        uint64
+	ExecutableBlock    *entity.ExecutableBlock
+	StateSnapshots     []*delta.SpockSnapshot
+	StateCommitments   []flow.StateCommitment
+	Proofs             [][]byte
+	Events             []flow.Event
+	ServiceEvents      []flow.Event
+	TransactionResults []flow.TransactionResult
+	GasUsed            uint64
+	StateReads         uint64
+	lock               sync.Mutex
+}
+
+func (cr *ComputationResult) AddEvents(inp []flow.Event) {
+	cr.lock.Lock()
```

Comment:
Not sure why we would need sync here? All those data must be ordered and recorded with transactional order - using mutex doesn't guarantee this, and if those data is written in parallel (I don't see it?) - other mechanism should be used to guarantee the above

Reply:
we probably don't need a sync here, I was not sure if we update proofs and states at the same time it can cause an issue or not?

Reply:
seems it is safe to update different members of the struct but since it will be part of the same cache line, I'll make it a one-time update instead.

Reply:
fixed